elastica/boundary_conditions.py

```diff
@@ -308,8 +308,20 @@ def __init__(self, *fixed_data, **kwargs):
             else:
                 # TODO: This part is prone to error.
                 break
-        self.fixed_positions = np.array(pos)
-        self.fixed_directors = np.array(dir)
+
+        if len(pos) > 0:
+            # transpose from (blocksize, dim) to (dim, blocksize)
+            self.fixed_positions = np.array(pos).transpose((1, 0))
+        else:
+            # initialize empty fixed positions
+            self.fixed_positions = np.zeros((3, 0))
+
+        if len(dir) > 0:
+            # transpose from (blocksize, dim, dim) to (dim, dim, blocksize)
+            self.fixed_directors = np.array(dir).transpose((1, 2, 0))
+        else:
+            # initialize empty fixed directors
+            self.fixed_directors = np.zeros((3, 3, 0))
```

Comment:
```suggestion
```

Reply:
Removed in https://github.com/GazzolaLab/PyElastica/pull/147/commits/c57b88d9cec9e8590f9305365acc6fe59026b022